scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -1673,9 +1715,18 @@ def _get_containers(
                 # Note we need to *include* results that were created after the 'running_or_created_after' time.
                 # that means we need to *ignore* any containers created before that
                 # hence the reason 'create_before' is assigned to a value named '...created_after'
-                if _ignore_old_dead_container(
+                is_old_dead_container = _ignore_old_dead_container(
                     container, created_before=running_or_created_after
-                ):
+                )
+
+                # AGENT-373/270 - we don't want to ignore containers that are still
+                # pending first use, otherwise we will miss out on short-lived logs
+                pending_first_use = False
+                if controlled_warmer is not None:
+                    pending_first_use = controlled_warmer.pending_first_use(cid)
+
+                # ignore containers that are old and dead, and that have been seen by the agent at least once
```

Comment:
How about "ignore containers that are old and dead and whose logs are already being copied"